pioinstaller/penv.py

```diff
@@ -86,6 +135,20 @@ def create_virtualenv_with_download(python_exe, penv_dir):
     command = [python_exe, venv_path, penv_dir]
     log.debug("Creating virtual environment: %s", " ".join(command))
     subprocess.check_output(command)
+    add_state_info(python_exe, penv_dir)
```

Comment:
Why do you call `add_state_info(python_exe, penv_dir)` in the different places? Why we can't call it in the one place "create_virtual_env"? When everything went OK.